src/persist/src/postgres.rs

```diff
@@ -263,6 +267,12 @@ impl PostgresConsensus {
     async fn get_connection(&self) -> Result<Object, PoolError> {
         let start = Instant::now();
         let res = self.pool.get().await;
+        if let Err(PoolError::Backend(err)) = &res {
+            // sadly, there does not appear to be an easy way to match on the error kind
+            if err.to_string().contains("connection timed out") {
```

Comment:
Might be able to dispatch on the `Error::source`, if there is one? But I think this is fine too.

Reply:
there's not an easy way to handle `source` since the inner type isn't pub. I decided to simplify this and just count the total # of errors we see during connection, I wasn't able to trigger a timeout locally to confirm the logic I had here to parse out timeouts was correct

Reply:
Pragmatic!